nixos/modules/services/networking/vpnc.nix

```diff
@@ -0,0 +1,97 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+
+  buildVpnc = key: value: {
+    name = "vpnc/${key}.conf";
+    value = {
+      text = ''
+        IPSec gateway ${value.gateway}
+        IPSec ID ${value.id}
+        IPSec secret ${value.secret}
+        Xauth username ${value.username}
+        Password helper ${pkgs.systemd}/bin/systemd-ask-password
+        Script ${pkgs.vpnc}/etc/vpnc/vpnc-script
+        '';
+    };
+  };
+  up = pkgs.writeScript "up" ''
+    #!${pkgs.bash}/bin/bash
```

Comment:
stdenv.shell

Reply:
Even better is `pkgs.runtimeShell`. `stdenv.shell` is not correct when cross-compiling